h2o-genmodel/src/main/java/hex/genmodel/easy/error/CountingErrorConsumer.java

```diff
@@ -13,24 +13,62 @@
  */
 public class CountingErrorConsumer extends EasyPredictModelWrapper.ErrorConsumer {
 
-  private AtomicLong dataTransformationErrorsCount = new AtomicLong();
-  private Map<String, AtomicLong> unknownCategoricalsPerColumn = new ConcurrentHashMap<>();
+  private Map<String, AtomicLong> dataTransformationErrorsCountPerColumn;
+  private Map<String, AtomicLong> unknownCategoricalsPerColumn;
 
   /**
    * @param model An instance of {@link GenModel}
    */
   public CountingErrorConsumer(GenModel model) {
+    initializeDataTransformationErrorsCount(model);
+    initializeUnknownCategoricalsPerColumn(model);
+  }
+
+  /**
+   * Initializes the map of data transformation errors for each column that is not related to response variable,
+   * excluding response and offset column. The map is initialized as unmodifiable and thread-safe.
+   *
+   * @param model {@link GenModel} the data trasnformation errors count map is initialized for
+   */
+  private void initializeDataTransformationErrorsCount(GenModel model) {
+    int predictorColumnsCount = model.isSupervised() ? model.getNames().length - 1 : model.getNames().length;
```

Comment:
This too complex IMHO. The initialization should be simple as simple as possible and the logic bulletproof. We should keep in mind that people will not have this always on and might enable it just for debugging. We don't want to introduce issue because of an ErrorConsumer.

I would keep it simple: for supervised models use all columns except for response and for unsupervised use all columns.

Excluding the offset column is conceptually not correct IMO because the offset column can also be incorrectly set (imagine a string instead of a number in RowData) and can have a transformation error.

Reply:
Removed offset column. Changed JavaDoc not to mention the offset column.

Removed the complex size calculation "in advance".